regression-tests.recursor-dnssec/recursortests.py

```diff
@@ -782,7 +783,9 @@ def tearDownAuth(cls):
 
     @classmethod
     def tearDownRecursor(cls):
-        cls.killProcess(cls._recursor)
+        p = cls.killProcess(cls._recursor)
+        if p.returncode not in (0, -15):
```

Comment:
Do you have any idea what is taking so long? If I understand the change above correctly, this means the recursor did not exit 10s after getting a term signal, which seems unexpected, and I wouldn't want to hide real issues like a thread stuck in a loop.

Reply:
`killProcess` sends a SIGTERM. Only after that fails it sends a SIGKILL.
I added the -15 case as there's no signal handler installed if no san is enabed. The SIGTerm then does the default action: kill the process, and Python gets a -15 as returncode.

Reply:
Understood, and do you know why after sending `SIGTERM` and waiting for a long time the recursor does not stop?

Reply:
I have not observed the SIGKILL case. With tsan _exiting can take more than a second on my local machine (the difference with this PR is `_exit()` is called, as opposed to a signal without a handler, the former is slower, the latter looks immediate), but I don't think I have observed the SIGKILL case after increasing the maximum timeout. I increased the max time a lot since GH runners can be slow. Maybe I'm too generous...

Reply:
I have the feeling I'm missing something...

SIGTERM not resulting in process exit is a problem. We do send SIGKILL then which is reported as a test failure by the PR (`returncode` will be -9).

Reply:
You are not, I am! This is a clear case of `-ENEEDCOFFEE`.